### PR TITLE
Eduroam source more flexible

### LINKED ISSUE #6948: Eduroam: make authentication source more flexible
**Is your feature request related to a problem? Please describe.**
On PF 11.2, when you create an Eduroam source, you can't:
- configure three Eduroam servers. Web admin lets you configure only two servers
- configure each Eduroam server with a different secret. Web admin lets you configure only one shared secret for all servers

I'm working with an UK university and configuration is generated automatically from Eduroam platform. They can't change secrets to use the same with all Eduroam servers.

I don't think creating additional Eduroam sources as a workaround will work because Eduroam sources are exclusive.

**Describe the solution you'd like**
* Be able to configure more than two servers on Eduroam sources. Would be nice to use same mechanism as for AD source when you can add several "hosts"
* Be able to specify a different secret for each Eduroam server

### PR DESCRIPTION
**Pull request by fdurand**
# Description

Reworked eduroam configuration in order to be able to define more than 2 eduroam servers.

# Impacts
Eduroam

# Issue
Fixes #6948

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Enhancements
* Unlimited eduroam radius servers for eduroam source

# UPGRADE file entries
Execute the script addons/upgrade/to-12.1-eduroam-migration.pl to migrate the configuration

### CODE REVIEW

File: docs/installation/authentication_mechanisms.asciidoc [hunk 1]
```diff
@@ -263,9 +263,15 @@ image::internal-eduroam-authentication.png[scaledwidth="100%",alt="Internal Edur
 
 Open the PacketFence administration web interface and go to _Configuration -> Policies and Access Control -> Authentication Sources_.
 
-Local *Exclusive Sources* and click on `New exclusive source` then `Eduroam`.
+First create as much RADIUS sources as you have Eduroam servers to define.
```
Comment (julsemaan) [inline docs/installation/authentication_mechanisms.asciidoc]:
```suggestion
First create RADIUS sources for each Eduroam servers you want to define.
```

File: html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm [hunk 2]
```diff
@@ -22,67 +22,43 @@ use HTML::FormHandler::Moose;
...
-    default         => pf::Authentication::Source::EduroamSource->meta->get_attribute('server2_port')->default,
-);
+has_field 'eduroam_radius_auth' =>
+  (
+   type => 'Select',
+   multiple => 1,
+   label => 'Eduroam RADIUS AUTH',
+   options_method => \&options_radius,
+   element_class => ['chzn-select'],
+   element_attr => {'data-placeholder' => 'Click to select a RADIUS Server'},
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm]:
```suggestion
```

File: html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm [hunk 3]
```diff
@@ -22,67 +22,43 @@ use HTML::FormHandler::Moose;
...
-    element_attr    => {
-        placeholder     => pf::Authentication::Source::EduroamSource->meta->get_attribute('server2_port')->default,
-    },
-    default         => pf::Authentication::Source::EduroamSource->meta->get_attribute('server2_port')->default,
-);
+has_field 'eduroam_radius_auth' =>
+  (
+   type => 'Select',
+   multiple => 1,
+   label => 'Eduroam RADIUS AUTH',
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm]:
```suggestion
```

Comment (julsemaan) [reply html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm]: That line has no purpose anymore now that everything is defined in the frontend

File: html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm (hunk 2 above)
Comment (julsemaan) [reply html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm]: That line has no purpose anymore now that everything is defined in the frontend

File: html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm [hunk 4]
```diff
@@ -22,67 +22,43 @@ use HTML::FormHandler::Moose;
...
+   options_method => \&options_radius,
+   element_class => ['chzn-select'],
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm]:
```suggestion
```

Comment (julsemaan) [reply html/pfappserver/lib/pfappserver/Form/Config/Source/Eduroam.pm]: That line has no purpose anymore now that everything is defined in the frontend

File: html/pfappserver/root/src/views/Configuration/sources/_components/FormTypeEduroam.vue [hunk 5]
```diff
@@ -14,19 +14,19 @@
       :column-label="$i18n.t('Description')"
     />
 
-    <form-group-server-address-port :namespaces="['server1_address', 'server1_port']"
-      :column-label="$i18n.t('Server 1 address')"
-      :text="$i18n.t('Eduroam server 1 address.')"
+    <form-group-eduroam-options namespace="eduroam_options"
+      :column-label="$i18n.t('Eduroam Realm Options')"
+      :text="$i18n.t('You can add Eduroam FreeRADIUS options in the realm definition.')"
```
Comment (julsemaan) [inline html/pfappserver/root/src/views/Configuration/sources/_components/FormTypeEduroam.vue]:
```suggestion
      :text="$i18n.t('FreeRADIUS options to add in the realm definition.')"
```

**Review by julsemaan — COMMENTED** [review]
Mostly verbiage changes.

Doc is adjusted and upgrade script is provided so for me this is good to merge

**Comment by fdurand** [comment]
Tested in real environment with incoming/outgoing request.
Ready on my side.

**Review by JeGoi — APPROVED** [review]
Looks fine to me

**Pull request merged**